openedx/core/djangoapps/content_libraries/views.py

```diff
@@ -713,8 +713,13 @@ def get(self, request, usage_key_str):
         Get the block's OLX
         """
         key = LibraryUsageLocatorV2.from_string(usage_key_str)
+        version = request.query_params.get('version', None)
```

Comment:
I think that we should actually move this API endpoint to the XBlock REST API.

After this:
https://github.com/openedx/edx-platform/blob/9cade7a3488eee24109311cb145747a01348df14/openedx/core/djangoapps/xblock/rest_api/urls.py#L20-L21

I didn't do this originally because we needed to check the `CAN_VIEW_THIS_CONTENT_LIBRARY` permission. But now we have `check_permission=CheckPerm.CAN_READ_AS_AUTHOR` available, so we can implement this in the XBlock API. I think that would be much nicer, and then in the future we can access the OLX for all XBlocks consistently, not just library blocks.

Please keep this libraries "draft" OLX REST API in place for backwards compatibility but mark it as deprecated and to be removed post-Teak.

Reply:
I would like to understand this approach well. Can we create another entry point, 'fields/olx,' and use `load_block` to get the block but only return the olx? In this approach, we need the user.

```
load_block(usage_key, request.user, check_permission=CheckPerm.CAN_READ_AS_AUTHOR, version=version)
```

Reply:
Ah, right - we don't want to use `load_block` at all, because we don't need to involve the runtime or actually load the block. It would be something like this:

```python
def get(self, request, usage_key: UsageKeyV2, version: LatestVersion | int = LatestVersion.AUTO):
    context_impl = get_learning_context_impl(usage_key)
    if not context_impl.can_view_block_for_editing(user, usage_key):
        raise PermissionDenied(f"You don't have permission to access the OLX of component '{usage_key}'.")
    olx = get_block_olx(usage_key, version)
    return Response(XBlockOlxSerializer({"olx": olx}).data)
```

Reply:
Also, it's fine to require a user for the REST API - in fact, we need to check the user's permissions. It's the python API, `get_block_draft_olx` / `get_block_olx` that shouldn't take a `user` argument, because it doesn't check permissions and the OLX doesn't vary per user.

Reply:
Updated https://github.com/openedx/edx-platform/pull/35932/commits/a66890d1cdc0c87464fbc15d5dbf8ebbd0ea38b2